### PR TITLE
Init auth

### PR DESCRIPTION
**Pull request by Lxkas**

### CODE REVIEW

**Review by chinathaip — APPROVED** [review]

File: apps/web/lib/supabase/component.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { SupabaseClient } from '@supabase/auth-helpers-nextjs'
+import { createBrowserClient } from '@supabase/ssr'
+
+// TODO: properly type this
+export function createClient(): SupabaseClient<any, "public", any> {
```
Comment (chinathaip) [inline apps/web/lib/supabase/component.ts:5]: later chai mai? XD

**Review by chinathaip — COMMENTED** [review]
Gucci. 

Obviously, we'll need to create another task to fix the header in mobile mode though (the size of  "Course Compose" should be adjusted accordingly, and the rest of header items should disappear)

File: apps/web/lib/supabase/component.ts (hunk 1 above)
Comment (Lxkas) [reply apps/web/lib/supabase/component.ts:5]: image

**Comment by NayHtetKyaw** [comment]
> Frontend gangs, do you have any inputs? @NayHtetKyaw @CryoMilo

no sirr. IT LOOKS GOOD

File: apps/web/components/core/application-header.tsx [hunk 2]
```diff
@@ -1,17 +1,115 @@
...
+		} catch (error) {
+			console.error(error);
+		}
+	};
+
+	const signInWithAzure = async (): Promise<void> => {
+		if (!supabase?.auth) return;
+
+		try {
+			const result = await supabase.auth.signInWithOAuth({ provider: "azure" });
```
Comment (chinathaip) [inline apps/web/components/core/application-header.tsx:50]: Also can you add this so it redirects to the same page after authentication? (right now it always goes to the first page)

```ts
supabase.auth.signInWithOAuth({
	provider: "azure",
	options: { redirectTo: window.location.href }
})
```

File: apps/web/hooks/use-auth.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                const password = "password";
+
+                await emailPasswordSignIn(supabase, email, password);
+            } else {
+                await signInWithAzure(supabase);
+            }
+        } catch (error) {
+            console.error("Error during the sign-in process:", error);
+            setAuthError(error);
+        }
```
Comment (Lxkas) [inline apps/web/hooks/use-auth.ts:60]: note to self: use the right env var

**Review by chinathaip — APPROVED** [review]

**Pull request merged**